swap_meet/vendor.py

```diff
@@ -1,2 +1,66 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = [] if inventory is None else inventory
+
+    def add(self, item): 
+        self.inventory.append(item)
+        return item
+ 
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if their_item not in other_vendor.inventory or my_item not in self.inventory:
+            return False
+        
+        self.inventory.remove(my_item)
+        self.inventory.append(their_item)
+        other_vendor.inventory.remove(their_item)
+        other_vendor.inventory.append(my_item)
+        return True
+
+        
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+
+        return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
```

Comment:
Nice reuse of a function you already wrote! However, notice that due to the remove calls in swap_items, this will be O(n) where n is the lengths of the inventories, since we'll have to shift the whole remaining part of the inventories forward. 

If you write your own implementation here, you could do this swap in constant time.

There is a way to swap elements in a list using multivariable assignment in python. That would look like: 
```python
a_list = ["cat", "dog"]
a_list[0], a_list[1] = a_list[1], a_list[0]
```

So for this method it would be like: 
```python
self.inventory[0], other_vendor.inventory[0] = other_vendor.inventory[0], self.inventory[0] 
```

Why do you think that swapping this way has less time complexity than using the remove() method? Would be good for you to investigate! 

Do you think the swapping syntax I showed above could be used in swap_items?